src/store/Chat/index.js

```diff
@@ -161,9 +180,34 @@ function ChatReducer (state = initialState, action) {
       //   });
       // }
 
+      // We need to run through the entire array of messages and aggregate them
+      //  into similar sets
+
+      if (action.payload.length > 0) {
+        sortedPayload = action.payload.sort((curr, next) => (
+          new Date(curr.timestamp) - new Date(next.timestamp)
+        ));
+        // There should be an algorithm here that would speed things up
```

Comment:
I think you're talking about the aggregation of messages from the same author... ? My motivation to do this on client/operator is because its purely for presentation, and the server/DB should not handle presentation bits. 

Buf if you were talking about the sortedPayload sort function I think there's a future improvement where the resulting messages are at least sorted on the server/DB side so that doesnt need to be done on client side.